### PR TITLE
Created Core Models

### PR DESCRIPTION
**Pull request by ElijahCaine**
fixes issue #18

## Changes in this PR.

Currently includes:
- Client
- Project
- Plugin
- NodeResource

Also adds rules to Rakefile for:
- Running app.rb
- Running migrations.
- `rake` and `rake test` depend on migrate

And I updated the docker setup:
- app_username's default command is `rake`
- dev_username's defualt command is `rake test`
- Added an `app.env.dist` file which includes DATABASE_URL.

## Testing this PR (in the dev environment)

```
# setup all of your environment
$ cp docker-compose.yml.dist docker-compose.yml  # Replace ALLCAPS variables
$ cp dockerfiles/app.dist dockerfiles/app  # Replace ALLCAPS variables
$ cp dockerfiles/app.env.dist dockerfiles/app.env  # should work as is
# Log into your dev instance and navigate to the project directory.
# Delete dev.rb if it already exists.
$ export DATABASE_URL=sqlite://dev.db
$ rake migrate
Migrating to latest
$ rake
/usr/local/bin/ruby app.rb
Clients:
GxxVIlZamx
Projects:
JTbDnOcLIv
NodeResources:
lTCjQLQpeS
Plugins:
== Sinatra (v1.4.7) has taken the stage on 4567 for development with backup from Thin
Thin web server (v1.6.4 codename Gob Bluth)
Maximum connections set to 1024
Listening on localhost:4567, CTRL+C to stop
```

@osuosl/devs @Kennric

### CODE REVIEW

**Comment by alxngyn** [comment]
this is pretty similar to #26 
I think we should include the migrate task into my rakefile since I also have rubocop and docker setup there.

**Comment by Kennric** [comment]
@alxngyn - actually, lets not put more into this, lets keep things in small pieces and merge/rebase as needed - when this is merged, the Rakefile migrate task can go on top of this

+1

**Comment by Kennric** [comment]
@ElijahCaine - just noticed that your models file is in the migrations directory - I don't think this is correct, models should be on the top level.

**Comment by alxngyn** [comment]
This is a stupid question, are we doing: 
base model -> migrate from here 
or 
empty models -> migrate from here
?

+1 from me either way.

File: migrations/001_migrations.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    create_table(:plugins) do
+      primary_key :id
+      String      :name,        :unique => true
+      String      :resource_name
+      String      :storage_table
+      String      :units
+    end
+
+    create_table(:node_resoruces) do
```
Comment (alxngyn) [inline migrations/001_migrations.rb]: typo: should be `:node_resources`

**Comment by ElijahCaine** [comment]
@alxngyn @Kennric described it as this:
1. `models.rb` describes the models and their relationships (`client` has a one-to-many relationship with `projects`, etc)
2. `_____migrations.rb` describes how the tables look (`client` has an `id`, `name`, etc). This changes with each migration.

**Comment by ElijahCaine** [comment]
@Kennric @alxngyn @matthewrsj One more review of this please. I made it actually work.

**Comment by alxngyn** [comment]
Anybody else getting this error when running: rake test?
I got it before the recent commit too.

> /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:2181:in block in set_restricted': method project_id= doesn't exist (Sequel::MassAssignmentRestriction)
>
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:2168:in each'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:2168:in set_restricted'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:1597:in set'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:2124:in initialize_set'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:1199:in initialize'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:170:in new'
> 	from /home/pvector/.rvm/gems/ruby-2.3.0/gems/sequel-4.33.0/lib/sequel/model/base.rb:170:in create'
> 	from app.rb:26:in <class:Iam>
> 	from app.rb:8:in <main
> rake aborted!

**Comment by Kennric** [comment]
+1

**Comment by matthewrsj** [comment]
+1 I'm not seeing an error

**Pull request merged**